src/proto/h1/role.rs

```diff
@@ -29,44 +29,37 @@ const AVERAGE_HEADER_SIZE: usize = 30; // totally scientific
 
 macro_rules! header_name {
     ($bytes:expr) => {{
-        #[cfg(debug_assertions)]
         {
             match HeaderName::from_bytes($bytes) {
                 Ok(name) => name,
-                Err(_) => panic!(
-                    "illegal header name from httparse: {:?}",
-                    ::bytes::Bytes::copy_from_slice($bytes)
-                ),
-            }
-        }
-
-        #[cfg(not(debug_assertions))]
-        {
-            match HeaderName::from_bytes($bytes) {
-                Ok(name) => name,
-                Err(_) => panic!("illegal header name from httparse: {:?}", $bytes),
+                Err(e) => maybe_panic!(e),
             }
         }
     }};
 }
 
 macro_rules! header_value {
     ($bytes:expr) => {{
-        #[cfg(debug_assertions)]
         {
             let __hvb: ::bytes::Bytes = $bytes;
             match HeaderValue::from_maybe_shared(__hvb.clone()) {
                 Ok(name) => name,
-                Err(_) => panic!("illegal header value from httparse: {:?}", __hvb),
+                Err(e) => maybe_panic!(e),
             }
         }
+    }};
+}
 
-        #[cfg(not(debug_assertions))]
-        {
-            // Unsafe: httparse already validated header value
-            unsafe { HeaderValue::from_maybe_shared_unchecked($bytes) }
```

Comment:
I just realized this change, which will makes things a little slower even when not using debug_assertions. Can we keep this code?

Reply:
Absolutely, seems like `from_maybe_shared_unchecked` internally only panics in debug builds so that should be good enough?